src/NAnt.Core/Tasks/LoopTask.cs

```diff
@@ -429,9 +429,15 @@ public enum LoopTrim {
                         break;
                 }
             } finally {
-                // Restore all of the old property values
-                for (int nIndex = 0; nIndex < oldPropVals.Length; nIndex++) {
-                    Properties[_props[nIndex]] = oldPropVals[nIndex];
+                // Restore all of the old property values. If any of the old properties
+                // contained a null value, remove that property from the dictionary.

```

Comment:
Only thing I might change is this comment to make it clear that the old property would be null because it simply didn't exist before the loop - the loop created it?